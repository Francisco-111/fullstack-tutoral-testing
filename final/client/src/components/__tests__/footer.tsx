// src/components/__tests__/footer.test.tsx
import React from "react";
import { shallow } from "enzyme";
import Footer from "../footer";

// If LogoutButton is exported and used directly inside Footer, you can import it:
// import LogoutButton from "../logout-button";

describe("<Footer />", () => {
    it("renders nav links", () => {
        const wrapper = shallow(<Footer />);
        const text = wrapper.text();
        expect(text).toContain("Home");
        expect(text).toContain("Cart");
        expect(text).toContain("Profile");
    });

    it("renders a logout control", () => {
        const wrapper = shallow(<Footer />);

        // Option A: If there's a LogoutButton component:
        // expect(wrapper.find(LogoutButton).exists()).toBe(true);

        // Option B: Fallback to text-based check if LogoutButton isn't directly importable:
        const text = wrapper.text().toLowerCase();
        expect(text).toMatch(/logout/); // or the componentâ€™s label if different
    });

    it("matches snapshot", () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper).toMatchSnapshot();
    });
});
