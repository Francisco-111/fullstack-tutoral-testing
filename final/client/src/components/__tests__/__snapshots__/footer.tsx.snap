// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `ShallowWrapper {}`;

exports[`<Footer /> matches snapshot 1`] = `ShallowWrapper {}`;
